### PR TITLE
Changed date guessing heuristic to use last modification time of posts instead of current time, if no better data is available.

### PR DESCRIPTION
**Pull request by kodfodrasz**
Changed date guessing heuristic to use last modification time of posts instead of current time, if no better data is available.

This is a viable and better replacement as all filesystems track this, and git and other version control tools also keep track of this and set it on files on checkout.

The changes:

 - Instead of DateTime.Now the source file's LastWriteTime is used now
 - The tests were updated
 - Updated System.IO.Abstractions and TestingHelpers to version 2.0.0.116
   to avoid a bug in the System.IO.Abstractions.TestingHelpers library
   making tests to fail
   (see: https://github.com/tathamoddie/System.IO.Abstractions/issues/122 )

### CODE REVIEW

File: src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs [hunk 1]
```diff
@@ -791,21 +807,25 @@ public void page_with_false_date_in_title()
         [Fact]
         public void post_default_values()
         {
-            fileSystem.AddFile(@"C:\TestSite\_posts\SomeFile.md", new MockFileData(@"---
+            var filename = @"C:\TestSite\SomeFile.md";
```
Comment (laedit) [inline src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs]: Why the change from Posts to Pages?

File: src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs [hunk 2]
```diff
@@ -787,8 +787,8 @@ public override LiquidEngine Given()
 
             public override void When()
             {
-                FileSystem.AddFile(@"C:\website\_site\BadFormat.md", new MockFileData(OriginalPageContents) { LastWriteTime = new DateTime(2010, 01, 2) });
-                FileSystem.AddFile(@"C:\website\BadFormat.md", new MockFileData(PageContents) { LastWriteTime = new DateTime(2010, 01, 3) });
+                FileSystem.AddFile(@"C:\website\_site\BadFormat.md", new MockFileData(OriginalPageContents) { LastWriteTime = new DateTime(2013, 01, 2) });
```
Comment (laedit) [inline src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs]: Why the year modification? Do you wanted to test different year values?

**Comment by laedit** [comment]
Good one, thanks a lot!
I just have some question about the tests modifications.

File: src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs (hunk 1 above)
Comment (kodfodrasz) [inline src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs]: Because the test failed. The test failed because the Posts get populated from _posts, Pages gets populated from anywhere in the site source. The path in the test vectors was not in the _posts folder (which made me wonder why the tests didn't fail earlier...). Instead of wasting hours, i changed it.

I hope I have not misunderstood something.

File: src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs (hunk 2 above)
Comment (kodfodrasz) [inline src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs]: I don't know. It may have come from some silly idea. I have cherry picked this commit from a dormant branch I was using since about half a year ago.

File: src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs (hunk 1 above)
Comment (laedit) [inline src/Pretzel.Tests/Templating/Context/SiteContextGeneratorTests.cs]: Well the file path was modified from "C:\TestSite\_posts\SomeFile.md" to "C:\TestSite\SomeFile.md" so it's normal that Posts is empty, but why change the file path?

File: src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs (hunk 2 above)
Comment (laedit) [inline src/Pretzel.Tests/Templating/Jekyll/LiquidEngineTests.cs]: Since it is not used for the tests, could you revert the changes?

**Comment by kodfodrasz** [comment]
I have addressed your comments. They were totally rightful. Sorry for them, I have just recently pulled some of my changes together from my sandbox branch, and decided to send them. This PR was only sandbox-quality.

**Comment by laedit** [comment]
No problem, thanks again for your work! :smiley:
